src/occupation.jl

```diff
@@ -3,67 +3,39 @@
 import Roots
 
 """
-Find Fermi level and occupation for the given parameters, assuming a band gap
-and zero temperature.
+Find the Fermi level, given a `basis`, SCF band `energies` and corresponding Bloch
+one-particle wave function `Psi`.
 """
-function find_occupation_gap_zero_temperature(basis, energies, Psi)
-    @assert basis.model.spin_polarisation in (:none, :spinless)
-    n_bands = size(Psi[1], 2)
-    n_electrons = basis.model.n_electrons
-    T = eltype(basis.kpoints[1].coordinate)
-    @assert basis.model.temperature == 0.0
-
-    filled_occ = filled_occupation(basis.model)
-    n_fill = div(n_electrons, filled_occ)
-    @assert filled_occ * n_fill == n_electrons
-    @assert n_bands ≥ n_fill
-
-    # We need to fill n_fill states with occupation filled_occ
-    # Find HOMO and LUMO
-    HOMO = -Inf # highest occupied energy state
-    LUMO = Inf  # lowest unoccupied energy state
-    occupation = similar(basis.kpoints, Vector{T})
-    for ik in 1:length(occupation)
-        occupation[ik] = zeros(T, n_bands)
-        occupation[ik][1:n_fill] .= filled_occ
-        HOMO = max(HOMO, energies[ik][n_fill])
-        if n_fill < n_bands
-            LUMO = min(LUMO, energies[ik][n_fill + 1])
-        end
-    end
-    @assert sum(basis.kweights .* sum.(occupation)) ≈ n_electrons
-
-    # Put Fermi level slightly above HOMO energy, to ensure that HOMO < εF
-    εF = nextfloat(HOMO)
-    if εF > LUMO
-        @warn("`find_occupation_gap_zero_temperature` assumes an insulator, but the " *
-              "system seems metallic. Try specifying a temperature and a smearing function.",
-              HOMO, LUMO, maxlog=5)
-    end
-
-    εF, occupation
+function find_fermi_level(basis, energies)
+    εF, _ = find_occupation(basis, energies)
+    εF
 end
 
 """
```

Comment:
My original idea here was to have `find_occupation`, which only returns the occupation, `find_fermi_level`, which only returns the Fermi level and `find_occupation_fermi`, which does both. You might disagree, but I find it unintuitive, why `find_occupation` also returns the Fermi level and it does so as first argument.

Reply:
Ah, I see. The reason I removed it is because I thought the `_fermi` had something to do with Fermi broadening, rather than the Fermi level. `find_occupation_fermi_level` is a bit long... The function is literally called once in the code, so I'm fine with whatever name; in general I think it's OK to return more than asked for; we could just switch the return values: return occupation, eps_F?

Reply:
Or use a named tuple ... there the "additional" returned object is kind of transparent. You can either use it or not.

Reply:
OK, let's do that. With tuple destructuring it's exactly the same for the caller.